### PR TITLE
AI Extensions: Check for null and undefined blocks

### PR DESCRIPTION
**Pull request by nunyvega**
Fixes https://github.com/Automattic/p2/issues/6150 , where  `is select( 'core/block-editor' ).getBlocksByClientId( clientIds ); `returns an array with null values, and is breaking lists and certain following blocks on the p2 comments box

## Proposed changes:

* Add a safeguard check for null and undefined values.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1687946664807469-slack-C054LN8RNVA

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Apply this patch in your sandbox
* Point a p2 page to your sandbox and confirm that the error does not occur anymore
* Test it on a simple site in the editor

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts [hunk 1]
```diff
@@ -84,7 +84,10 @@ export function getTextContentFromSelectedBlocks(): GetTextContentFromBlocksProp
 		count: blocks.length,
 		clientIds,
 		content: blocks
-			? blocks.map( block => getBlockTextContent( block.clientId ) ).join( HTML_JOIN_CHARACTERS )
+			? blocks
+					.filter( block => block !== null && block !== undefined ) // Safeguard against null or undefined blocks
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts:88]: Should we make the condition stronger, checking, for instance, the name object?
```suggestion
					.filter( block => !! block?.name ) // Safeguard against null or undefined blocks
```

**Review by retrofox — APPROVED** [review]
Left a suggestion. Feel free to ignore it. Nice fix!

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts (hunk 1 above)
Comment (dhasilva) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts:88]: `block !== null && block !== undefined` is the same as `block != null`

**Comment by samiff** [comment]
cherry-picked to `monthly/branch-2023-06-26` in 39badef8fa34c37a8daf055e13851dbc883ce8fb

**Pull request merged**